exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -476,10 +476,26 @@ func addSingleDoubleSummaryDataPoint(pt *otlp.DoubleSummaryDataPoint, metric *ot
 	}
 }
 
+func orderBySampleTimestamp(tsArray []prompb.TimeSeries) []prompb.TimeSeries {
+	for i := range tsArray {
+		sL := tsArray[i].Samples
+		sort.Slice(sL, func(i, j int) bool {
+			return sL[i].Timestamp < sL[j].Timestamp
+		})
+		tsArray[i].Samples = sL
+	}
+	return tsArray
+}
+
 func convertTimeseriesToRequest(tsArray []prompb.TimeSeries) *prompb.WriteRequest {
 	// the remote_write endpoint only requires the timeseries.
 	// otlp defines it's own way to handle metric metadata
 	return &prompb.WriteRequest{
-		Timeseries: tsArray,
+		// First things first, we need to ensure that the samples per time series are sorted by Timestamp.
+		// This ensures that we don't produce out of order problems
```

Comment:
I think it is better to avoid comments like "we do x":

// Prometheus required time series to be sorted, in order to ensure no out of order data.
// See:
// ...

Reply:
Done, thank you @bogdandrutu!